src/Shared/PackageManagers/PackageManagerFactory.cs

```diff
@@ -28,50 +30,39 @@ public static BaseProjectManager CreateBaseProjectManager(IHttpClientFactory htt
         /// <returns></returns>
         public static BaseProjectManager CreateBaseProjectManager(string destinationDirectory)
         {
-            return new BaseProjectManager(destinationDirectory);
+            return new BaseProjectManager(new BaseProvider(), destinationDirectory);
```

Comment:
If the intent here is to have a provider that complies with the interface but does not actually return any data, perhaps either a) the BaseProjectManager could be modified to allow the provider to be nullable, returning NotSupported if any relevant calls are made, or b) a NoOpProvider could be added instead of relying on functionality in a base class that does the same thing as the base class right now, but can instead be sealed so that no other classes try to extend it.

Reply:
The only reference in the lib I can find to this is from DetectCryptographyTool. I too am not quite sure why you would want this interface instead of the one based on purl.

Reply:
Removed